### ISSUE TITLE
Tab groups (like in Chromium)

### ISSUE THREAD

**Issue by rebelcode-xr** [pulsar-edit/pulsar#510]
### Have you checked for existing feature requests?

- [X] Completed

### Summary

Would be great to have "tab groups" like in Chromium/Chrome:

<img width="365" alt="image" src="https://user-images.githubusercontent.com/63522951/235363324-ce2ed71e-66e3-4901-8347-1d48eccd94ae.png">

With the ability to add a **name** and a **colour** to a group of tabs in just few seconds.
The group can also shrink/expand, which is also very convenient: IMO this should be a must feature since Pulsar is based on Chromium.

### What benefits does this feature provide?

For visual people like me, the ability to group the tabs and color them would help keep track of where I am when I have many tabs open. Since I need to jump from one to another I frequently lose myself. The way they are implemented in Chromium makes them very easy to create/change/delete a group in few seconds, so they can adapt to different scenarios.

### Any alternatives?

I tried some packages to colour tabs but they don't solve my problem: I use Brave to navigate the web, which is based on Chromium, so I know the feature very well and there's nothing like that right now.

**Comment by Daeraxa** [comment]
It seems this is a pretty popular request (yet to be implemented) for VSCode too:

https://github.com/microsoft/vscode/issues/100335

An alternative (as mentioned in that issue thread) would be Vivaldi style tab stacking - https://vivaldi.com/fr/whats-new-in-vivaldi-3-6/

Honestly I personally think this would be a nice idea, I often want to keep a ton of stuff open on a single window but want to swap project and this seems a nice way to accomplish it.

**Comment by savetheclocktower** [comment]
Since tab themselves are implemented as a package, this is certainly possible to do as a third-party package, so maybe a Pulsar fan would like to take a stab at it.

**Comment by confused-Techie** [comment]
@savetheclocktower I wonder, was just looking at the tabs package, is it possible to modify the tabs from that package easily? Since I know theres no service registered as a provider for tabs, but I wonder how difficult that might be. Such as hooking into keyboard shortcuts and the like.

Since really, this does bring up the question of what else can be done with tabs, such as the possibility of both Tab grouping, or tab stacking. Or even minimizing tabs similar to how Chromium based browsers do, or even such a thing as pinned tabs, color coded. There's tons of options we could explore

**Comment by confused-Techie** [comment]
Although now that I said this, taking a look there are dozens of community packages that aim to accomplish many of these.

* [`stacked-tabs`](https://web.pulsar-edit.dev/packages/stacked-tabs) - Display tabs stacked, just like Sublime Text.
* [`pinned-tabs`](https://web.pulsar-edit.dev/packages/pinned-tabs) - Pin a tab, inspired by browsers.
* [`color-tabs-byproject`](https://web.pulsar-edit.dev/packages/color-tabs-byproject) - Color tabs based on path or subfolder. Probably the most similar to what's being requested here.

So the options here are endless, and this very very likely could be implemented as an additional community package, and seeing the amazing work community packages do in this space, there's no reason this has to be a core feature

**Comment by rebelcode-xr** [comment]
Sadly those were made for Atom and are pretty old and buggy:
[stacked-tabs](https://web.pulsar-edit.dev/packages/stacked-tabs) latest release is 6 years old
[pinned-tabs](https://web.pulsar-edit.dev/packages/pinned-tabs) latest release is 3 years old
[color-tabs-byproject](https://web.pulsar-edit.dev/packages/color-tabs-byproject) latest release is 4 years old

This is a core feature in most browsers right now, since Pulse uses Chromium why not having it activated? I mean...it should be already in the code, right?

**Comment by confused-Techie** [comment]
@rebelcode-xr Not really. Electron does use the chromium engine, but from my understanding, that only relates to rendering web content. Which tabs, the top bar of Chrome (such as your bookmark bar, the URL bad, and extensions area) is all specific to chromium itself, but it is not part of the chrome engine.

So to implement this, it would have to be coded from scratch to appear similar to chromes implementation. It wouldn't just be 'activating' something

**Comment by rebelcode-xr** [comment]
ohhh 😰 I understand, that's a pity. thanks for the explanation!

**Comment by confused-Techie** [comment]
@rebelcode-xr Yeah wish it could be so easy lol. But of course, but unfortunate to hear those existing community packages are buggy. But at the very least hopefully those can provide some groundwork for either new community plugins, or if this was implemented as a core feature. Thanks for letting me know they aren't preforming the best

**Comment by rebelcode-xr** [comment]
Actually I found out that was another package to create incompatibility: [rainbow-tabs-evolved](https://web.pulsar-edit.dev/packages/rainbow-tabs-evolved)
Maybe there's already somebody working on implementing this behaviour for Electron/Chrome engine... 🤔

**Comment by savetheclocktower** [comment]
Leaning toward closing this unless folks think that tab groups are a feature that's likely to make it into core. If someone really wants this feature, I would encourage them to fork `tabs` and publish an enhanced version as a community package.
